Release 1.20.0 (#418)
scrapinghub/spidermon: CHANGES.rst

Release notes
=============

1.20.0 (2023-09-01)
-------------------

> - feature: Add support for nested fields in ``SPIDERMON_VALIDATION_ERRORS_FIELD`` setting (`PR#416 <https://github.com/scrapinghub/spidermon/pull/417>`_)
> - bug: Fix ItemValidationPipeline using ``__setitem__`` method without ItemAdaper (`PR#415 <https://github.com/scrapinghub/spidermon/pull/415>`_)
> - feature: Improve tests coverage for item validation pipeline (`PR#414 <https://github.com/scrapinghub/spidermon/pull/414>`_) 
> - bug: Fix ``spidermon.utils.zyte.Client`` not following crawler settings (`PR#411 <https://github.com/scrapinghub/spidermon/pull/411>`_)
> - chore: Migrate ReadTheDocs to config file v2 (`PR#408 <https://github.com/scrapinghub/spidermon/pull/408>`_)
> - chore: Refactor package requirements (`PR#407 <https://github.com/scrapinghub/spidermon/pull/407>`_)

Reviewer: ```suggestion
- feature: Add support for nested fields in the ``SPIDERMON_VALIDATION_ERRORS_FIELD`` setting (`PR#416 <https://github.com/scrapinghub/spidermon/pull/417>`_)
- bug: Fix ItemValidationPipeline using the ``__setitem__`` method without ItemAdaper (`PR#415 <https://github.com/scrapinghub/spidermon/pull/415>`_)
- feature: Improve test coverage for the item validation pipeline (`PR#414 <https://github.com/scrapinghub/spidermon/pull/414>`_) 
- bug: Fix ``spidermon.utils.zyte.Client`` not following crawler settings (`PR#411 <https://github.com/scrapinghub/spidermon/pull/411>`_)
- chore: Migrate ReadTheDocs to config file v2 (`PR#408 <https://github.com/scrapinghub/spidermon/pull/408>`_)
- chore: Refactor package requirements (`PR#407 <https://github.com/scrapinghub/spidermon/pull/407>`_)
```

> Backwards incompatible changes in 1.20.0
> ----------------------------------------

Reviewer: ```suggestion
Backward-incompatible changes in 1.20.0
---------------------------------------
```

Or shorter:

```suggestion
Backward-incompatible changes
-----------------------------
```

Author: I prefer the first version because the title is at the same indentation level as the 1.20.0 release notes. So, in the future there won't be multiples `Backward-incompatible changes` entries at the same level.

Reviewer: I did not notice the level, and I would suggest nesting it to 1.20.0 instead, but no strong opinion.

> - Keys that contains dots can't not longer be used as field name to store validation errors in dictionaries due to changes to ``SPIDERMON_VALIDATION_ERRORS_FIELD``.

Author: I'm not proud of my wording here, but I can't explain it better. Some help is very much appreciated

Reviewer: What about…

```suggestion
- The ``SPIDERMON_VALIDATION_ERRORS_FIELD`` setting no longer supports a field 
  name that contains a dot (``.``). Dots are now be treated as field 
  separators, to support specifying a nested field.  
```
